pkg/k8shandler/certificates.go

```diff
@@ -45,6 +50,72 @@ func extractSecretToFile(namespace string, secretName string, key string, toFile
 	return nil
 }
 
+func writeSecret(logging *v1alpha1.ClusterLogging) (err error) {
+
+	secret := utils.Secret(
+		"logging-master-ca",
+		logging.Namespace,
+		map[string][]byte{
+			"masterca":   utils.GetFileContents("/tmp/_working_dir/ca.crt"),
+			"masterkey":  utils.GetFileContents("/tmp/_working_dir/ca.key"),
+			"kibanacert": utils.GetFileContents("/tmp/_working_dir/kibana-internal.crt"),
+			"kibanakey":  utils.GetFileContents("/tmp/_working_dir/kibana-internal.key"),
+		})
+
+	utils.AddOwnerRefToObject(secret, utils.AsOwner(logging))
+
+	err = sdk.Create(secret)
+	if err != nil && !errors.IsAlreadyExists(err) {
```

Comment:
Isn't it better to return early if the secret exists already?

Reply:
if it exists we still want to make sure the certs aren't expired, if they were then we need to update the secret because we generated new ones.